controllers/front/listing/ManufacturerController.php

```diff
@@ -239,8 +239,10 @@ public function getTemplateVarPage()
     {
         $page = parent::getTemplateVarPage();
 
-        $page['body_classes']['manufacturer-id-' . $this->manufacturer->id] = true;
-        $page['body_classes']['manufacturer-' . $this->manufacturer->name] = true;
+        if (Validate::isLoadedObject($this->manufacturer) && $this->manufacturer->active && $this->manufacturer->isAssociatedToShop()) {
```

Comment:
```suggestion
        if (!empty($this->manufacturer->id)) {
```

I agree with @PululuK. There's a check for that on the `init` method, so you only need to check whether you're on the manufacturer page or the listing, and you can do that by checking the object. No need for checking association, etc., because it is checked already at this point.